master/internal/api_experiment.go

```diff
@@ -936,6 +937,33 @@ func (a *apiServer) PatchExperiment(
 	return &apiv1.PatchExperimentResponse{Experiment: exp}, nil
 }
 
+type Checkpoint struct {
+	bun.BaseModel `bun:"table:bun_checkpoints_view"`
+	Uuid          string
+	TaskId        string
+	AllocationId  string
+	ReportTime    time.Time
+	State         string
+	Resources     map[string]int64
+	Metadata      *structpb.Struct
+	Training      *checkpointv1.CheckpointTrainingMetadata
+	ExperimentId  int32
+	TrialId       int32
+}
+
+func (c Checkpoint) Proto() *checkpointv1.Checkpoint {
+	return &checkpointv1.Checkpoint{
+		TaskId:       c.TaskId,
+		AllocationId: c.AllocationId,
+		Uuid:         c.Uuid,
+		ReportTime:   timestamppb.New(c.ReportTime),
+		Resources:    c.Resources,
+		Metadata:     c.Metadata,
+		State:        checkpointv1.State(checkpointv1.State_value[c.State]),
+		Training:     c.Training,
+	}
+}
+
```

Comment:
these should probably go somewhere else. not sure if this PR is big enough to warrant starting the modularization process though.